test/cases/notifications_test.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+require "abstract_unit"
+require "fixtures/person"
+
+class NotificationsTest < ActiveSupport::TestCase
+  def setup
+    matz = { person: { id: 1, name: "Matz" } }
+    @people = { people: [ matz ] }.to_json
+
+    ActiveResource::HttpMock.respond_to do |mock|
+      mock.get  "/people.json?name=Matz", { "Accept" => "application/json" }, @people
+      mock.post "/people.json", { "Content-Type" => "application/json" }, nil, 201, "Location" => "/people/5.json"
+    end
+  end
+
+  def test_get_request_with_params
+    payload = capture_notifications { Person.where(name: "Matz") }
+
+    assert_equal :get, payload[:method]
+    assert_equal "http://37s.sunrise.i:3000/people.json?name=Matz", payload[:request_uri]
+    assert_equal({ "Accept" => "application/json" }, payload[:headers])
+    assert_nil payload[:body]
+    assert_kind_of ActiveResource::Response, payload[:result]
```

Comment:
This assertion is a quirk of how `ActiveResource::HttpMock.respond_to` works. In reality, the `payload[:result]` kind is a `Net::HTTPResponse`, but in the test harness it's an `ActiveResource::Response`.